tests/ui/cast/ptr-to-ptr-different-regions.rs

```diff
@@ -0,0 +1,16 @@
+// check-pass
+
+// https://github.com/rust-lang/rust/issues/113257
+
+struct Foo<'a> { a: &'a () }
+
+fn extend_lifetime_very_very_safely<'a>(v: *const Foo<'a>) -> *const Foo<'static> {
+    // This should pass because raw pointer casts can do anything they want.
+    v as *const Foo<'static>
+}
```

Comment:
Could you add something like my example from https://github.com/rust-lang/rust/issues/114798 to this test, to make sure it also works for dyn trait pointer lifetime bounds?